ledger/block/src/transaction/deployment/mod.rs

```diff
@@ -181,6 +222,45 @@ impl<N: Network> Deployment<N> {
     }
 }
 
+impl<N: Network> Deployment<N> {
+    /// Sets the program checksum.
+    /// Note: This method is intended to be used by the synthesizer **only**, and should not be called by the user.
+    #[doc(hidden)]
+    pub fn set_program_checksum_raw(&mut self, program_checksum: Option<[U8<N>; 32]>) {
+        self.program_checksum = program_checksum;
+    }
+
+    /// Sets the program owner.
+    /// Note: This method is intended to be used by the synthesizer **only**, and should not be called by the user.
+    #[doc(hidden)]
+    pub fn set_program_owner_raw(&mut self, program_owner: Option<Address<N>>) {
+        self.program_owner = program_owner;
+    }
+
+    /// An internal function to return the implicit deployment version.
+    /// This function implicitly checks that the deployment checksum and owner is well-formed.
+    #[doc(hidden)]
+    pub(super) fn version(&self) -> Result<DeploymentVersion> {
```

Comment:
Can this be made public?

Reply:
Can be public if there's a need to use this version.

Also, why are the `set_*_raw` functions `pub` instead of `pub(super)` if we want to restrict usage?